core/src/main/java/org/apache/calcite/sql/parser/SqlParserUtil.java

```diff
@@ -353,6 +353,30 @@ public static long intervalToMillis(
         interval.getIntervalQualifier());
   }
 
+  public static long weekIntervalToMillis(
+      SqlIntervalLiteral.IntervalValue interval) {
+    return weekIntervalToMillis(
+        interval.getIntervalLiteral(),
+        interval.getIntervalQualifier());
+  }
+
+  public static long weekIntervalToMillis(
+      String literal,
+      SqlIntervalQualifier intervalQualifier) {
+
+    int[] ret;
+    try {
+      ret = intervalQualifier.evaluateIntervalLiteral(literal,
+          intervalQualifier.getParserPosition(), RelDataTypeSystem.DEFAULT);
+      assert ret != null;
+    } catch (CalciteContextException e) {
+      throw new RuntimeException("TODO WEEK INTERVAL MESSAGE HERE "
+          + literal, e);
```

Comment:
This seems like it needs to be fixed before merging

Reply:
Not relevant anymore since I'm refactoring "WEEK" to be handled directly inside of `intervalToMillis` now instead.